vkcs/mlplatform/resource_jupyterhub.go

```diff
@@ -0,0 +1,567 @@
+	Region           types.String              `tfsdk:"region"`
+	Timeouts         timeouts.Value            `tfsdk:"timeouts"`
+}
+
+func (r *JupyterHubResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Instance name. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"domain_name": schema.StringAttribute{
+				Required:    true,
+				Description: "Domain name. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"admin_name": schema.StringAttribute{
+				Optional:    true,
+				Description: "JupyterHub admin name. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"admin_password": schema.StringAttribute{
+				Optional:    true,
+				Sensitive:   true,
+				Description: "JupyterHub admin password. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"flavor_id": schema.StringAttribute{
+				Required:    true,
+				Description: "Flavor ID",
+			},
+
+			"availability_zone": schema.StringAttribute{
+				Required:    true,
+				Description: "The availability zone in which to create the resource. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"boot_volume": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"size": schema.Int64Attribute{
+						Required:    true,
+						Description: "Size of the volume",
+					},
+					"volume_type": schema.StringAttribute{
+						Required:    true,
+						Description: "Type of the volume",
+					},
+					"name": schema.StringAttribute{
+						Computed:    true,
+						Description: "Name of the volume",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+					"volume_id": schema.StringAttribute{
+						Computed:    true,
+						Description: "ID of the volume",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+				Required:    true,
+				Description: "Instance's root volume configuration",
+			},
+
+			"data_volumes": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"size": schema.Int64Attribute{
```

Comment:
could we specify this set once to reuse it?

Reply:
moved common schema attributes to shared.go